plugin/src/resolve-gatsby-image-data.ts

```diff
@@ -1,21 +1,43 @@
 import {
   generateImageData,
+  getLowResolutionImageURL,
   IGatsbyImageHelperArgs,
   IImage,
   ImageFormat,
 } from "gatsby-plugin-image";
+import fetch from "node-fetch"
 import { ShopifyImage, urlBuilder } from "./get-shopify-image";
 
 type ImageLayout = "constrained" | "fixed" | "fullWidth";
 
+type IImageWithPlaceholder = IImage & {
+  placeholder: string
+}
+
+async function getImageBase64({ imageAddress }: { imageAddress: string }): Promise<string> {
+  const response = await fetch(imageAddress)
+  const buffer = await response.buffer();
+  return buffer.toString(`base64`)
+}
+
+/**
+ * Download and generate a low-resolution placeholder
+ * 
+ * @param lowResImageFile
+ */
+function getBase64DataURI({ imageBase64 }: { imageBase64: string }) {
+  return `data:image/png;base64,${imageBase64}`
+}
+
 export async function resolveGatsbyImageData(
   image: Node & ShopifyImage,
   {
     formats = ["auto"],
     layout = "constrained",
     ...options
-  }: { formats: Array<ImageFormat>; layout: ImageLayout }
+  }: { formats: Array<ImageFormat>; layout: ImageLayout; options: { placeholder: string } }
```

Comment:
You probably don't need this type definition anymore, now that we are casting the `remainingOptions`.

Reply:
Tested and confirmed - don't need it! Removed :+1: